src/java.base/share/classes/java/lang/reflect/Method.java

```diff
@@ -230,13 +230,13 @@ public int getModifiers() {
 
     /* package */
     Optional<?> setCodeModelIfNeeded(Function<Method, Optional<?>> modelFactory) {
-        Optional<?> localRef = codeModel;
+        Optional<?> localRef = root.codeModel;
```

Comment:
I was looking at `Method` and found this method that does more or less a similar operation:

```
// Accessor for generic info repository
    @Override
    MethodRepository getGenericInfo() {
        var genericInfo = this.genericInfo;
        if (genericInfo == null) {
            var root = this.root;
            if (root != null) {
                genericInfo = root.getGenericInfo();
            } else {
                genericInfo = MethodRepository.make(getGenericSignature(), getFactory());
            }
            this.genericInfo = genericInfo;
        }
        return genericInfo;
    }
```

Should we copy/reuse same logic?

Reply:
E.g. note that `root` can be null!